### PR TITLE
feat: intial hudi reg test

### PR DESCRIPTION
**Pull request by rahil-c**
## Summary
* Adding intial regression test for polaris hudi integration, following exist pattern set by Delta regression test
* Made changes in run.sh and setup.sh in order to ensure that spark session can be started correctly depending on the table format.
* Ran locally both delta regression test and hudi regression test to ensure they pass.

cc @gh-yzou @singhpk234 @flyrain 

## Checklist
- [ ] 🛡️ Don't disclose security issues! (contact security@apache.org)
- [ ] 🔗 Clearly explained why the changes are needed, or linked related issues: Fixes #
- [ ] 🧪 Added/updated tests with good coverage, or manually tested (and explained how)
- [ ] 💡 Added comments for complex logic
- [ ] 🧾 Updated `CHANGELOG.md` (if needed)
- [ ] 📚 Updated documentation in `site/content/in-dev/unreleased` (if needed)

### CODE REVIEW

File: plugins/spark/v3.5/regtests/spark_hudi.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# Add TABLE_WRITE_DATA to the catalog's catalog_admin role since by default it can only manage access and metadata
+curl -i -X PUT -H "Authorization: Bearer ${SPARK_BEARER_TOKEN}" -H 'Accept: application/json' -H 'Content-Type: application/json' \
+  http://${POLARIS_HOST:-localhost}:8181/api/management/v1/catalogs/${CATALOG_NAME}/catalog-roles/catalog_admin/grants \
+  -d '{"type": "catalog", "privilege": "TABLE_WRITE_DATA"}' > /dev/stderr
+
+curl -H "Authorization: Bearer ${SPARK_BEARER_TOKEN}" -H 'Accept: application/json' -H 'Content-Type: application/json' \
+  "http://${POLARIS_HOST:-localhost}:8181/api/catalog/v1/config?warehouse=${CATALOG_NAME}"
+echo
+echo "Catalog created"
+cat << EOF | ${SPARK_HOME}/bin/spark-sql -S \
```
Comment (dimas-b) [inline plugins/spark/v3.5/regtests/spark_hudi.sh]: Would it be possible to run this test as an Integration test under JUnit inside the Gradle builld?

Comment (rahil-c) [reply plugins/spark/v3.5/regtests/spark_hudi.sh]: Hi @dimas-b we currently have the following integration test for polaris hudi here: https://github.com/apache/polaris/pull/3194

In terms of the reg test, I was the following the shell pattern that @gh-yzou had done for Delta, now for Hudi.

Comment (gh-yzou) [reply plugins/spark/v3.5/regtests/spark_hudi.sh]: Hi @dimas-b The purpose of this regression test is to validate the end-to-end user experience when using Spark with both --packages and --jars. This is an important scenario that cannot be fully covered by integration tests.
While it is true that this test is relatively expensive, that is why it includes only very basic test cases. More complex scenarios and edge cases are covered by integration tests, which provide a more cost-effective approach.

File: plugins/spark/v3.5/regtests/run.sh [hunk 2]
```diff
@@ -70,6 +70,21 @@ SPARK_VERSION="3.5.6"
 
 SPARK_SHELL_OPTIONS=("PACKAGE" "JAR")
 
+# Define test suites to run
+# Each suite specifies: test_file:table_format:test_shortname
+declare -a TEST_SUITES=(
+  "spark_sql.sh:delta:spark_sql"
```
Comment (gh-yzou) [inline plugins/spark/v3.5/regtests/run.sh]: How about let's enforce the test file name to the format like xxx_<table_format>.sh, and have a separate folder to include all test src file and reference file. Then we just need to list the folder to get all test files, and extract the table format by parsing the file name. The benefit would be easy to onboard new tests, and developer doesn't have to input a long string when running single test (just the file name)

File: plugins/spark/v3.5/regtests/setup.sh [hunk 3]
```diff
@@ -182,7 +232,12 @@ fi
 echo "Launch spark-sql at ${SPARK_HOME}/bin/spark-sql"
 # bootstrap dependencies so that future queries don't need to wait for the downloads.
 # this is mostly useful for building the Docker image with all needed dependencies
-${SPARK_HOME}/bin/spark-sql -e "SELECT 1"
+if [[ "$TABLE_FORMAT" == "hudi" ]]; then
+  # For Hudi: Pass --packages on command line to match official Hudi docs approach
```
Comment (gh-yzou) [inline plugins/spark/v3.5/regtests/setup.sh]: i don't think we need the if else here anymore

File: plugins/spark/v3.5/regtests/spark_hudi.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+drop table iceberg_tb;
+drop namespace hudi_db1;
+drop namespace hudi_db2;
+EOF
+
+# clean up the spark_hudi_catalog dir
+rm -rf /tmp/spark_hudi_catalog/
+
+curl -i -X DELETE -H "Authorization: Bearer ${SPARK_BEARER_TOKEN}" -H 'Accept: application/json' -H 'Content-Type: application/json' \
+  http://${POLARIS_HOST:-localhost}:8181/api/management/v1/catalogs/${CATALOG_NAME} > /dev/stderr
```
Comment (gh-yzou) [inline plugins/spark/v3.5/regtests/spark_hudi.sh]: new line

File: plugins/spark/v3.5/regtests/run.sh [hunk 5]
```diff
@@ -70,6 +70,48 @@ SPARK_VERSION="3.5.6"
...
+# Auto-discover test suites from the suites/ directory
+# Test files must follow naming convention: <name>_<table_format>.sh
+SUITES_DIR="${SCRIPT_DIR}/suites"
+
+if [[ ! -d "$SUITES_DIR" ]]; then
+  logred "Error: Test suites directory not found: ${SUITES_DIR}"
+  exit 1
+fi
+
+parse_test_suite() {
```
Comment (gh-yzou) [inline plugins/spark/v3.5/regtests/run.sh:86]: can we add a comment here about what this function is doing, it is trying to extract the TABLE_ROMAT, TEST_SHORTNAME and the full path of TEST_FILE, right?

File: plugins/spark/v3.5/regtests/run.sh [hunk 6]
```diff
@@ -70,6 +70,48 @@ SPARK_VERSION="3.5.6"
...
+  [[ -f "$test_file" ]] || continue
+  TEST_SUITES+=("$(basename "$test_file")")
+done
+
+if [[ ${#TEST_SUITES[@]} -eq 0 ]]; then
+  logred "Error: No test suites found in ${SUITES_DIR}"
+  exit 1
+fi
+
+# Allow running specific test via environment variable
```
Comment (gh-yzou) [inline plugins/spark/v3.5/regtests/run.sh:105]: I think we can potentially also allow running all suites for a particular format by taking table format as an argument to this script. We can probably do that in a separate PR as an improvement.

**Review by gh-yzou — DISMISSED** [review]

File: plugins/spark/v3.5/regtests/spark_hudi.sh (hunk 1 above)
Comment (dimas-b) [reply plugins/spark/v3.5/regtests/spark_hudi.sh]: I believe Spark-based scenarios can be covered in regular CI with sufficient certainty... but I do not mean to block this PR on this point :slightly_smiling_face: Please consider optional.

File: plugins/spark/v3.5/regtests/run.sh (hunk 6 above)
Comment (rahil-c) [reply plugins/spark/v3.5/regtests/run.sh:105]: lets do in another pr

File: plugins/spark/v3.5/regtests/run.sh (hunk 5 above)
Comment (rahil-c) [reply plugins/spark/v3.5/regtests/run.sh:86]: ack

**Review by flyrain — APPROVED** [review]
